redshift_sqlalchemy/dialect.py

```diff
@@ -320,67 +323,157 @@ def visit_unload_from_select(element, compiler, **kw):
-                null - Optional string value denoting what to interpret as a NULL value from the file
-                manifest - Boolean value denoting whether data_location is a manifest file; defaults to False
-                empty_as_null - Boolean value denoting whether to load VARCHAR fields with
-                                empty values as NULL instead of empty string; defaults to True
-                blanks_as_null - Boolean value denoting whether to load VARCHAR fields with
-                                 whitespace only values as NULL instead of whitespace; defaults to True
-        '''
-        self.schema_name = schema_name
-        self.table_name = table_name
+
+class CopyCommand(Executable, ClauseElement):
+    """
+    Prepares a Redshift COPY statement.
+
+    Parameters
+    ----------
+    table : sqlalchemy.Table
+        The table to copy data into
+    data_location : str
+        The Amazon S3 location from where to copy, or a manifest file if
+        the `manifest` option is used
+    credentials : str
+        AWS credentials of the form
+        aws_access_key_id=<access-key-id>;aws_secret_access_key=<secret-access-key>[;token=<temporary-session-token>]
+    delimiter : File delimiter, optional
+        defaults to ','
+    ignore_header : int, optional
+        Integer value of number of lines to skip at the start of each file
+    null : str, optional
+        Optional string value denoting what to interpret as a NULL value from
+        the file
+    manifest : bool, optional
+        Boolean value denoting whether data_location is a manifest file.
+    empty_as_null : bool, optional
+        Boolean value denoting whether to load VARCHAR fields with empty
+        values as NULL instead of empty string
+    blanks_as_null : bool, optional
+        Boolean value denoting whether to load VARCHAR fields with whitespace
+        only values as NULL instead of whitespace
+    format : str, optional
+        CSV, JSON, or AVRO. Indicates the type of file to copy from.
+    compression : str, optional
+        GZIP, LZOP, indicates the type of compression of the file to copy
+    """
+    def __init__(self, table, data_location, credentials,
+                 delimiter=',', ignore_header=0, null=r'\N', manifest=False,
+                 empty_as_null=True, blanks_as_null=True,
+                 format='CSV', compression=None):
+        # regex from here: http://blogs.aws.amazon.com/security/blog/tag/key+rotation
+        if not creds_rx.match(credentials):
+            raise ValueError('"credentials" parameter must match the following'
+                             ' format:\n'
+                             'aws_access_key_id=<access-key-id>;'
+                             'aws_secret_access_key=<secret-access-key>'
+                             '[;token=<temporary-session-token>]\ngot %r' %
+                             credentials)
```

Comment:
why did you change the way the credentials are passed in? This seems way more brittle